src/core/unit.c

```diff
@@ -6473,6 +6470,18 @@ int unit_arm_timer(
         return 0;
 }
 
+bool unit_is_filtered(Unit *u, char *const *states, char *const *patterns) {
+        assert(u);
+
+        if (!strv_isempty(states) &&
+            !strv_contains(states, unit_load_state_to_string(u->load_state)) &&
+            !strv_contains(states, unit_active_state_to_string(unit_active_state(u))) &&
+            !strv_contains(states, unit_sub_state_to_string(u)))
```

Comment:
Please switch to `strv_overlap(states, STRV_MAKE(load_state, active_state, sub_state)` if you touch this